includes/Core/Authentication/Clients/Google_Proxy_Client.php

```diff
@@ -140,6 +141,37 @@ public function revokeToken( $token = null ) {
 		return 200 === (int) $response->getStatusCode();
 	}
 
+	/**
+	 * Adds auth listeners to the HTTP client based on the credentials set in the Google API Client object.
+	 *
+	 * @since 1.0.3
+	 *
+	 * @param ClientInterface $http The HTTP client object.
+	 * @return ClientInterface The HTTP client object
+	 */
+	public function authorize( ClientInterface $http = null ) {
+		if ( $this->isUsingApplicationDefaultCredentials() ) {
+			return parent::authorize( $http );
+		}
+
+		$token = $this->getAccessToken();
+		if ( isset( $token['refresh_token'] ) && $this->isAccessTokenExpired() ) {
+			$callback = $this->config['token_callback'];
+
+			try {
+				$creds = $this->fetchAccessTokenWithRefreshToken( $token['refresh_token'] );
+				if ( $callback ) {
+					// Due to original callback signature this can only accept the token itself.
+					call_user_func( $callback, '', $creds['access_token'] );
+				}
+			} catch ( \Exception $e ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement
+				// Ignore exceptions.
+			}
```

Comment:
This logic should never be hit because we manually refresh it in our codebase, but I wanted to fix this regardless. This entire change to the `Google_Proxy_Client` is pretty much made only for hypothetical case that we would wanna rely on it at some point, or through some edge case a request would be made with an expired access token.